app/controllers/application_controller.rb

```diff
@@ -315,6 +315,17 @@ def require_current_organization
     redirect_to new_sessions_organization_path(redirect_back_to: request.original_url)
   end
 
+  def require_not_on_waiting_list
+    return if !current_market.waiting_list_enabled?
+    return if current_user.try(:market_manager?)
+    return if adding_items_to_existing_order?
+
+    if current_organization.try(:on_waiting_list?)
```

Comment:
why use `try`? Just call it directly, there is a default value.

Reply:
What if `current_organization` is nil?